src/Mvc/Mvc.Core/src/Infrastructure/ProblemDetailsJsonConverter.cs

```diff
@@ -20,7 +20,7 @@ public override ProblemDetails Read(ref Utf8JsonReader reader, Type typeToConver
         {
             var problemDetails = new ProblemDetails();
 
-            if (!reader.Read())
+            if (reader.TokenType != JsonTokenType.StartObject)
```

Comment:
Noticed this when I wrote a functional test. The converter assumed the reader did not read the initial token which is incorrect when invoked by the serializer. Added a couple of tests.

Reply:
So we can safely assume the reader will never read t he initial token?

Reply:
For an JSON object, the serializer always reads the first token before calling the converter.

```
{ "myProp": "myValue" }
  ^ ---- is where the token is.
```

All the existing tests used the raw reader which is why we never caught this. FYI @ahsonkhan

Reply:
Yes. The deserializer already moved the reader state to StartObject.

For another example, see https://github.com/dotnet/corefx/blob/75d19b61b20197f3e6348f5e444e9e3af0f0750a/src/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonValueConverterKeyValuePair.cs#L20-L25